src/macaron/repo_finder/repo_finder_deps_dev.py

```diff
@@ -146,7 +146,13 @@ def get_latest_version(purl: PackageURL) -> tuple[PackageURL | None, RepoFinderI
         versions = json_extract(metadata, versions_keys, list)
         if not versions:
             return None, RepoFinderInfo.DDEV_JSON_INVALID
-        latest_version = json_extract(versions[-1], ["versionKey", "version"], str)
+
+        latest_version = None
+        for index, version_result in enumerate(versions):
+            if version_result["isDefault"] or (index == (len(versions) - 1) and not latest_version):
```

Comment:
Can `latest_version` not be `None` and reach here?

Reply:
Please add a comment in the code to explain the logic here.

Reply:
It is intended to accept the version that has the `isDefault` property set to `True`, or the final entry in the list in case one isn't found. However, if none are set to default, maybe it's better to log an error and return nothing as that represents some issue with `deps.dev` or our expectations.

Reply:
> It is intended to accept the version that has the `isDefault` property set to `True`, or the final entry in the list in case one isn't found. However, if none are set to default, maybe it's better to log an error and return nothing as that represents some issue with `deps.dev` or our expectations.

Thanks, I meant to just add that in code as a comment :slightly_smiling_face: